src/coreclr/jit/optcse.cpp

```diff
@@ -3129,16 +3152,25 @@ void CSE_HeuristicRLHook::GetFeatures(CSEdsc* cse, int* features)
             continue;
         }
 
-        if (!varTypeIsFloating(varTyp))
+        if (varTypeUsesIntReg(varTyp))
         {
-            enregCount++; // The primitive types, including TYP_SIMD types use one register
+            enregCountInt++;
```

Comment:
This is an example of where we would previously skip floating-point values but this would include `simd` and `mask`, despite `simd` using the same register set as `float`.

This meant the enregistration count would get off by a method utilizing SIMD, which is undesirable.

The new logic correctly tracks the enregistration count of the 3 different register types that exist today.